plugins/exec.lua

```diff
@@ -11,6 +11,19 @@ local function exec(cmd, keep_newline)
 end
 
 
+local function exec_rw(dv, cmd, keep_newline)
+  local proc = process.start { "sh", "-c", cmd }
+  proc:write(tostring(dv))
+  local res = ""
+  while true do
+    local rdbuf = proc:read_stdout()
+    if not rdbuf then break end
+    res = res .. rdbuf
+  end
+  return keep_newline and res or res:gsub("%\n$", "")
+end
```

Comment:
For performance reasons I'd suggest
```suggestion
local function exec_rw(dv, cmd, keep_newline)
  local proc = process.start { "sh", "-c", cmd }
  proc:write(tostring(dv))
  local res = {}
  while true do
    local rdbuf = proc:read_stdout()
    if not rdbuf then break end
    if #rdbuf > 0 then table.insert(res, rdbuf) end
  end
  res = table.concat(res)
  return keep_newline and res or res:gsub("%\n$", "")
end
```